cmd/fanoutsidecar/main.go

```diff
@@ -84,14 +81,18 @@ func main() {
 		logger.Fatal("--sidecar_port flag must be set")
 	}
 
+	if len(channelProvisioners) < 1 {
+		logger.Fatal("--channel_provisioners must be specified")
+	}
+
 	sh, err := swappable.NewEmptyHandler(logger)
 	if err != nil {
 		logger.Fatal("Unable to create swappable.Handler", zap.Error(err))
 	}
 
-	mgr, err := setupConfigMapNoticer(logger, sh.UpdateConfig)
+	mgr, err := setupChannelWatcher(logger, sh.UpdateConfig)
```

Comment:
The previous impl was also channel specific as the Swappable.UpdateConfig relies on multichannelfanout.Config which is crafted to be Channels specific. Hence I modified it such that other channels can use it if needed by passing in the provisioners. But something that is not channel specific cannot use it.
Moving away from configMap to directly reading the channel makes sure there is just one source of truth and lesser knobs that could potentially break ( no need for some other service to update the config map).
We could make it generic if needed. However as you noted other than in-memory channel nothing uses it. As we talked in-person, my opinion will be to remove dead code and keep it specific to channels. In case a need arises in future we can always go back to this revision and bring some code back.

Reply:
`multichannelfanout.Config` was _named_ with `Channel`, but is not inherently tied to Channels in anyway. It would be easy to craft a config to do arbitrary fanouts, without having any real Channels. So the existing code is generic, but has poorly named fields. 

That having been said, I am OK with removing the functionality. @evankanderson do you agree?

Reply:
Here is the consensus on the offline discussion (@evankanderson , @Harwayne , @akashrv). 
1. We'll keep the change in this PR as it i - watch channels directly and not through configmaps
2. Create a new issue to make fan-out side a library rather than a re-usable container.

Reply:
[Here is the issue](https://github.com/knative/eventing/issues/1053)

Reply:
To record a bit more context:

1. Do we still need the fan-out as a separate container/sidecar? Between work on the CloudEvents library to support different encoding schemes and need for different fanout strategies for different persistence mechanisms (see the most recent Kafka consumer changes, for example), I'm not sure that generic fanout will be useful when only Channels are performing fanout.

2. It sounds like the fanout sidecar is mostly used as the implementation of the `in-memory` Channel. If so, then renaming to reflect its current purpose makes sense.